plugins/targetmanagers/csvtargetmanager/csvfile.go

```diff
@@ -155,15 +156,30 @@ func (tf *CSVFileTargetManager) Acquire(jobID types.JobID, cancel <-chan struct{
 			hosts[i], hosts[j] = hosts[j], hosts[i]
 		})
 	}
-	if uint32(len(hosts)) > acquireParameters.MaxNumberDevices {
-		hosts = hosts[:acquireParameters.MaxNumberDevices]
+
+	// feed all devices into new API call `TryLock`, with desired limit
+	lockedString, err := tl.TryLock(jobID, hosts, uint(acquireParameters.MaxNumberDevices))
+	if err != nil {
+		return nil, fmt.Errorf("can't lock enough targets")
```

Comment:
The API only guarantees the upper limit, no minimum, it can return with zero targets but no error
```suggestion
		return nil, fmt.Errorf("failed to lock targets: %w", err)
```